compaction/group_id.hh

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2023-present ScyllaDB
+ *
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include <fmt/format.h>
+
+namespace compaction {
+
+struct group_id {
+    unsigned idx;
+    unsigned total;
+
+    group_id() noexcept : idx(0), total(0) {}
+    group_id(unsigned idx, unsigned total) noexcept : idx(idx), total(total) {}
+    explicit group_id(unsigned total) noexcept : idx(0), total(total) {}
+
+    group_id& operator++() noexcept {
+        ++idx;
+        return *this;
```

Comment:
an improvement for later is to check if we're going out of valid range and throw std::out_of_range if so.